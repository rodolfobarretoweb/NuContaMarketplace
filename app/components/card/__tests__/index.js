import React from 'react';
import { View } from 'react-native';
import renderer from 'react-native-test-utils';
import checkPropTypes, { assertPropTypes } from 'check-prop-types';
import Card from '../index';

describe('Load component', () => {
  it('renders snapshot correctly', () => {
    const card = renderer(<Card><View /></Card>);
    expect(card.toJSON()).toMatchSnapshot();
  });

  it('fails without children prop', () => {
    expect(() => {
      assertPropTypes(Card.propTypes, { children: undefined }, 'prop');
    }).toThrow();
  });

  it('fails with wrong prop type on `show` prop', () => {
    expect(
      checkPropTypes(Card.propTypes, { children: '' }, 'prop', 'Card')
    ).toContain('Invalid prop `children` of type `string` supplied to `Card`, expected a single ReactElement.');
  });
});
