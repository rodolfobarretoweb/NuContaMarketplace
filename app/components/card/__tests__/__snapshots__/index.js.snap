// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load component renders snapshot correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#fff",
        "elevation": 3,
        "marginBottom": 10,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0.1,
          "width": 1,
        },
        "shadowOpacity": 0.3,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View />
</View>
`;
